### DISCUSSION TITLE
upload of a single file with fastapi

### DISCUSSION THREAD

**Discussion by fcabaud** [fastapi/fastapi#7030] [Questions] [answered]
### First check

[ * ] I added a very descriptive title to this issue.
[ * ] I used the GitHub search to find a similar issue and didn't find it.
[ * ] I searched the FastAPI documentation, with the integrated search.
[ * ] I already searched in Google "How to X in FastAPI" and didn't find any information.
[ * ] I already read and followed all the tutorial in the docs and didn't find an answer.
[ * ] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
[ * ] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
[ * ] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
[ * ] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

I only tried to used this documentation for uploading a single file[:](https://fastapi.tiangolo.com/tutorial/request-files/)

```Python
from fastapi import FastAPI, Request, Form
from fastapi.templating import Jinja2Templates
from fastapi.staticfiles import StaticFiles
from fastapi import  File, UploadFile

@app.post("/import_file")
async def import_file_post(file: UploadFile = File(...)):
    logger.info('post import_file')
    return {"filename": file.filename}
```
with html data
```html
    <label for="myfile">Select a file:</label>
    <form action="/import_file/" enctype="multipart/form-data" method="post">
        <input name="files" type="file">
        <input type="submit">
    </form>
```

### Description

* Open the browser and call the endpoint `/`.
http://127.0.0.1:8000/import_file

When a click on `submit` on the browser I got this error 

`{"detail":[{"loc":["body","file"],"msg":"field required","type":"value_error.missing"}]} `

logs are 
```
INFO:     127.0.0.1:46044 - "POST /import_file/ HTTP/1.1" 307 Temporary Redirect
INFO:     127.0.0.1:46044 - "POST /import_file HTTP/1.1" 422 Unprocessable Entity

```

### Environment

* OS: [e.g. Linux Ubuntu 20.10]:
* FastAPI Version [e.g. 0.61.1]:

To know the FastAPI version use:

* Python version: 3.9.0+

### Additional context
I'm uploading an excel file.

**Comment by Mause** [comment, marked as answer]
Your name in your form is wrong - it should be `file`, or the argument to your endpoint should be `files`

**Comment by fcabaud** [comment]
Wow, you are right.
Thanks

**Comment by tiangolo** [comment]
Thanks for the help here @Mause ! :clap: :bow: 

Thanks for reporting back and closing the issue :+1:

**Comment by SadScheme** [comment]
Thanks for this issue i was dealing with the same problem too
